packages/expo-screen-orientation/CHANGELOG.md

```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- [ios] Fix crash when reading rootViewController value. ([#23039](https://github.com/expo/expo/pull/23039) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```

Comment:
```suggestion
- [iOS] Fix crash when reading `rootViewController` value. ([#23039](https://github.com/expo/expo/pull/23039) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```

Reply:
@tsapeta New code returns `EXRootViewController` and then `nil` while the popup is closing, in that case `screen-orientation` considers the supported interface orientations for that `ViewController` as an empty array and handles everything correctly.
While closing the popup `UIApplication.shared.keyWindow?.rootViewController` returns `UIApplicationRotationFollowingController` instead od the nil, it seems like `screen-orientation` has issues handling this. The issue looks like it's thread related . In all cases other than closing the popup output of the two functions seems to be te same

Reply:
Yeah, it's exactly what Wojciech said